src/charm.py

```diff
@@ -196,23 +181,25 @@ def _config_changed(self, event: ConfigChangedEvent):
         self.get_livepatch_snap.restart(["livepatch"])
 
         if self.unit.status.message == AWAIT_POSTGRES_RELATION:
-            event.defer()
+            if event is not None:
+                event.defer()
             return
 
         if self.livepatch_running is not True:
             self.set_status_and_log("Livepatch failed to restart.", MaintenanceStatus)
-            event.defer()
+            if event is not None:
+                event.defer()
         else:
             self._update_status(event)
 
-    def _update_status(self, event: UpdateStatusEvent) -> None:
+    def _update_status(self, _: Any) -> None:
         """Perform a simple service health check."""
         logging.info("Updating application status...")
         current_status = self.unit.status
         if self._check_install_and_relations():
             if self.livepatch_running:
                 self.set_status_and_log("Livepatch running!", ActiveStatus)
-            elif current_status == ActiveStatus:
+            elif isinstance(current_status, ActiveStatus):
```

Comment:
Why do we need this change?

Reply:
Because the comparison is actually wrong. We're comparing an object, `current_status`, with a type, `ActiveStatus`, which always results in `False`. Note that in the `set_status_and_log` method we receive the type as a parameter and then instantiate from it and assign it to the unit status:

https://github.com/canonical/livepatch-machine-charm/blob/ad9352c4cf8ca037ff2ce9a7a8b252b217cb6fb2/src/charm.py#L403-L406